src/components/TransactionPreview/HistoryPreview.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import styled, { useTheme } from 'styled-components';
+import { ethers } from 'ethers';
+
+// Components
+import Card from '../Card';
+import { ClickableText, Text } from '../Text';
+import { CombinedRoundedImages, RoundedImage } from '../Image';
+
+// Utils
+import { formatAmountDisplay, humanizeHexString, copyToClipboard, getTypeOfAddress } from '../../utils/common';
+import { Chain, CHAIN_ID, klimaAsset, nativeAssetPerChainId, supportedChains } from '../../utils/chain';
+import { Theme } from '../../utils/theme';
+
+// Constants
+import { TRANSACTION_BLOCK_TYPE } from '../../constants/transactionBuilderConstants';
+import { CHAIN_ID_TO_NETWORK_NAME } from 'etherspot/dist/sdk/network/constants';
+
+import moment from 'moment';
+
+// Hooks
+import { useEtherspot } from '../../hooks';
+
+// Types
+import { ICrossChainAction, SendAssetActionPreview } from '../../types/crossChainAction';
+
+//Icons
+import { RiExternalLinkLine } from 'react-icons/ri';
+import { LuFuel } from 'react-icons/lu';
+
+const MainWrapper = styled.div`
+  display: inline-block;
+  display: flex;
+  flex-direction: row;
+  background: ${({ theme }) => theme.color.background.card};
+  width: 100%;
+`;
+
+const LeftWrapper = styled.label`
+  font-size: 14px;
+  float: left;
+  text-align: left;
+  width: 50%;
+`;
+
+const RightWrapper = styled.label`
+  font-size: 14px;
+  float: right;
+  text-align: right;
+  width: 50%;
+`;
+
+const Label = styled.label`
+  display: inline-block;
+  color: ${({ theme }) => theme.color.text.innerLabel};
+  margin-bottom: 10px;
+  font-size: 14px;
+`;
+
+const DateLabel = styled.label`
+  display: inline-block;
+  color: ${({ theme }) => theme.color.text.innerLabel};
+  margin-right: 7px;
+  margin-bottom: 10px;
+  font-size: 14px;
+`;
+
+const ValueWrapper = styled.div<{ marginTop?: number }>`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  ${({ marginTop }) => marginTop && `margin-top: ${marginTop}px;`}
+`;
+
+interface TransactionPreviewInterface {
+  crossChainAction: ICrossChainAction;
+}
+
+const HistoryPreview = ({ crossChainAction }: TransactionPreviewInterface) => {
+  const { accountAddress, providerAddress, web3Provider } = useEtherspot();
+  const theme: Theme = useTheme();
+
+  const { chainId, type } = crossChainAction;
+
+  const openBlockExplorerUrl = (transactionUrl) => {
+    if (!transactionUrl) return;
```

Comment:
In what scenario would there not be a `transactionurl` - and is it something we need to let the user know about?

Reply:
We always get URL but I handled if not condition. I tested we always get explorerURL from SDK.

Reply:
Please don't assume the SDK works perfectly, it doesn't. If you're assuming that we may not get a `transactionUrl`, it needs to be handled gracefully

Reply:
changes done